controllers/storagesystem_controller.go

```diff
@@ -224,3 +225,35 @@ func (r *StorageSystemReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
 		Complete(r)
 }
+
+// RemoveSubscriptions deletes any managed Subscriptions that do not have any
+// existing CRs that their operators reconcile.
+func (r *StorageSystemReconciler) RemoveSubscriptions() error {
+	var err error
+
+	for _, kind := range KnownKinds {
+		subs := GetSubscriptions(kind)
+		for _, sub := range subs {
+			if clientErr := r.Client.Delete(context.TODO(), sub); clientErr != nil {
+				if err == nil {
+					err = clientErr
+				} else {
+					err = fmt.Errorf("%w; ", clientErr)
+				}
+			}
+		}
+	}
+
+	return err
+}
+
+// GetOwnerReferences returns an OwnerReference for the specified StorageSystem
+func GetOwnerReferences(instance *odfv1alpha1.StorageSystem) []metav1.OwnerReference {
```

Comment:
I would like to see this func before setupWithManager exactly after addReferenceToRelatedObjects. If possible let's covert this as a method on top of the reconciler.

Reply:
There's no need to do that. In fact, I was rather thinking of getting rid of this entirely since I found a function that does exactly this: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference

Reply:
Got rid of it! Turns out it was just an artifact of a temporary solution. :joy: